tests.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Default script used by the Docker testing harness in the container run by ./docker-tests.sh
+
+# Strict, and loud failure
+set -euo pipefail
+trap 'rc=$?;set +ex;if [[ $rc -ne 0 ]];then trap - ERR EXIT;echo 1>&2;echo "*** fail *** : code $rc : $DIR/$SCRIPT $ARGS" 1>&2;echo 1>&2;exit $rc;fi' ERR EXIT
+ARGS="$*"
+DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+SCRIPT="$(basename "${BASH_SOURCE[0]}")"
+
+set -x
+
+# Dev-centric test steps based on scripts in package.json
+PATH=$(npm bin):$PATH
+
+# If ava testing hangs (which, sadly, happens all too often), add --serial to get a more
```

Comment:
I haven't been seeing ava hang - do you have any repro steps? What's your system configuration?

Reply:
Heh, this PR captures my system config. ;-)  It's Docker for Mac.  Within this framework it's intermittent, and less likely when the `--serial` option is given to `ava`.   There are perhaps three tests that sometimes fail.  But, given ava's relative secrecy about what's being run, it's hard to figure out which test is hung.  I'm new to a lot of the tools you're using, so I'm a bit tentative about my conclusions.

 If I run Node from the command-line on the Mac (using the brew-installed Node 10), the tests almost never succeed..., that's what got me working on Docker and Bee-Queue.

Reply:
Yeah ava was an early adoption and while on the one hand I'd kinda prefer jest for its stability and support I do still like the raw speed that ava provides. If only the author of `ava-spec` didn't deprecate it in favor of Jest...

I might have to try and get my hands on a mac to repro what you're seeing - it'd be nice if these tests succeeded, and race conditions like the ones you're talking about concern me.

Reply:
I will spend a little time on looking at this PR/branch in a Docker system on an underlying Linux kernel (unlike Mac, which uses the moby kernel emulator or whatever you call it).

Reply:
Experiment:

I used the PR branch on a Digital Ocean droplet which has an Ubuntu-based Docker environment -- So, stable, cloud-centric Linux kernel and mainstream Ubuntu/Debian Docker.

I used this script, `npmtest.sh`:
```bash
#!/bin/sh                                                                                                                                                                                                                                

exec npm test
```
Run via:
```
(while ./docker-tests.sh npmtest.sh ; do echo && echo timestamp && date && echo ; done) > /tmp/beequeuetests.log 2>&1
```
On this Droplet a new "timestamp" line would appear in the log file every 22 seconds or so, until... either:
- the while loop to stopped, meaning an explicit test failure
- the number of "timestamp" lines in the log file  stopped increasing, meaning a hung test

Out of 75 execs of `npm test` I got:
- 67 successful runs
- 7 hugh tests
- 1 failure

So, overall 89% success rate.

The one failure (which I have occasionally seen on Mac):
```
bee-queue_1  |   139 passed
bee-queue_1  |   1 failed
bee-queue_1  |
bee-queue_1  |   helpers-test › finallyRejectsWithInitial waits for the returned Promise
bee-queue_1  |
bee-queue_1  |   /bee-queue/test/helpers-test.js:85
bee-queue_1  |
bee-queue_1  |    84:     );
bee-queue_1  |    85:     t.true(measure() >= 10);
bee-queue_1  |    86:     t.true(stub.calledOnce);
bee-queue_1  |
bee-queue_1  |   Value is not `true`:
bee-queue_1  |
bee-queue_1  |   false
bee-queue_1  |
bee-queue_1  |   measure() >= 10
bee-queue_1  |   => false
bee-queue_1  |
bee-queue_1  |   measure()
bee-queue_1  |   => 7.332423
bee-queue_1  | npm ERR! Test failed.  See above for more details.
bee-queue_bee-queue_1 exited with code 1
```
The 7 hung runs had all completed 139 of the tests.   I'll have to work a little to figure out which test(s) were the one(s) that hung.

Reply:
I have goodish news.  So far, using my uuid-based approach (#237) to fixing the Job.id race condition, I am not seeing any of the non-deterministic test failures and hangs.